product_variant_to_weigh_with_scale/views/product_view.xml

```diff
@@ -7,7 +7,21 @@
         <field name="inherit_id" ref="product_variant_available_in_pos.product_variant_easy_edit_view"/>
         <field name="arch" type="xml">
             <field name="available_in_pos" position="after">
-                <field name="to_weight" attrs="{'invisible': [('available_in_pos', '=', False)]}"/>
+                 <field name="uom_category_name" invisible="1"/>
+                <field name="to_weight" attrs="{'invisible': ['|', ('available_in_pos', '=', False), ('uom_category_name', '!=', 'Weight')]}"/>
+            </field>
+        </field>
+    </record>
+
+     <record id="product_template_form_view_inherit" model="ir.ui.view">
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="point_of_sale.product_template_form_view"/>
+        <field name="arch" type="xml">
+            <field name="available_in_pos" position="after">
+                <field name="uom_category_name" invisible="1"/>
+            </field>
+            <field name="to_weight" position="attributes">
+                <attribute name="attrs">{'invisible': ['|', ('available_in_pos', '=', False), ('uom_category_name', '!=', 'Weight')]}</attribute>
```

Comment:
The invisible attr on `('uom_category_name', '!=', 'Weight')` seems a bit janky, however i'm assuming there is no other way to flag a uom category as weight. Maybe we could check if it matches the xmlid `product_uom_categ_kgm`?